core/src/types.rs

```diff
@@ -172,6 +186,134 @@ pub enum TypedValue {
     Uuid(Uuid),                        // It's only 128 bits, so this should be acceptable to clone.
 }
 
+/// The values stored in a `StructuredMap` can be:
+///
+/// * Vecs of structured values, for multi-valued component attributes or nested expressions.
+/// * Single structured values, for single-valued component attributes or nested expressions.
+/// * Single typed values, for simple attributes.
+///   Note that the `TypedValue` enum is inlined here for convenience: `Binding` is
+///   an expansion of it.
+///
+/// Datomic also supports structured _bindings_; at present Mentat does not, but this type
+/// would also serve that purpose.
+///
+/// Note that maps are not ordered….
```

Comment:
It seems like you think this is significant.  If it is, explain why.

Reply:
… 'cos it means this doesn't implement `Ord`.